tests/plugins/test_pulp_pull.py

```diff
@@ -528,11 +530,121 @@ def get_manifest(request):
         workflow.postbuild_plugins_conf = [{'name': 'pulp_sync'}]
         tasker = MockerTasker()
         plugin = PulpPullPlugin(tasker, workflow, timeout=0.2,
-                                retry_delay=0.1, expect_v2schema2=True)
+                                retry_delay=0.1, expect_v2schema2=expect_v2)
         media_types = plugin.run()
 
         expected_media_types = [MEDIA_TYPE_DOCKER_V2_MANIFEST_LIST]
         if not manifest_list_only:
             expected_media_types.append(MEDIA_TYPE_DOCKER_V2_SCHEMA1)
             expected_media_types.append(MEDIA_TYPE_DOCKER_V2_SCHEMA2)
+        else:
+            assert "Only V2 schema 2 manifest list is expected, " in caplog.text()
         assert set(media_types) == set(expected_media_types)
+
+    @responses.activate
+    @pytest.mark.parametrize(('found_v2', 'found_v2_list'), [
+        (True, True),
+        (True, False),
+        (False, True),
+        (False, False),
+    ])
+    @pytest.mark.parametrize('expect_v2', [True, False])
+    @pytest.mark.parametrize(('platforms', 'manifest_list_only'), [
+        (['ppc64le'], True),
+        (['ppc64le', 'arm'], True),
+        (['ppc64le', 'x86_64'], False),
+        (['x86_64'], False),
+    ])
+    def test_expect_v2_expect_v2list(self, caplog, found_v2, found_v2_list,
+                                     expect_v2, platforms, manifest_list_only):
+        def get_manifest(request):
+            media_types = request.headers.get('Accept', '').split(',')
+            content_type = media_types[0]
+            return (200, {'Content-Type': content_type}, '{}')
+
+        url = re.compile(r'.*//crane.example.com/v2/.*/manifests/.*')
+        responses.add_callback(responses.GET, url, callback=get_manifest)
+        digests = {'digest': None}
+        workflow = self.workflow(
+            platform_descriptors=True,
+            postbuild_results={PLUGIN_GROUP_MANIFESTS_KEY: digests},
+            prebuild_results={PLUGIN_CHECK_AND_SET_PLATFORMS_KEY: set(platforms)},
+            expectv2schema2=expect_v2,
+        )
+        workflow.postbuild_plugins_conf = [{'name': 'pulp_sync'}]
+
+        not_found = requests.Response()
+        flexmock(not_found, status_code=requests.codes.not_found)
+        expectation = flexmock(requests.Session).should_receive('get')
+
+        # 1st retry don't find anything
+        for _ in range(5):
+            expectation = expectation.and_return(not_found)
+
+        expect_v1 = False
+        if 'x86_64' in platforms:
+            expect_v1 = True
+
+        # (for v1, v2, list.v2, oci, and oci.index media types) before get_manifest_digests
+        # v1
+        if expect_v1:
+            expectation.and_return(self.config_response_config_v1)
+        else:
+            expectation.and_return(not_found)
+
+        # v2
+        if found_v2:
```

Comment:
It is confusing to use variable `found_v2` to set up the expectation.

Reply:
Actually the terminology used in this test is ambiguous to me. What's meant by `found_*` vs `expect_*` ?

I think something like `expect_*` vs `has_*` is more straightforward.